File: frontend/vigie-du-mensonge/src/core/models/politician.ts
export type PoliticianJson = {
    id: string;
    fullName: string;
};

export class Politician {
    public id: string;
    public fullName: string;

    constructor(id: string, fullName: string) {
        this.id = id;
        this.fullName = fullName;
    }

    public static fromJson(json: PoliticianJson): Politician {
        return new Politician(json.id, json.fullName);
    }
}

File: backend/core/dto/response_dto/politician.go
package response_dto

import (
	"vdm/core/models"

	"github.com/google/uuid"
)

type Politician struct {
	ID       uuid.UUID `json:"id"`
	FullName string    `json:"fullName"`
}

func NewPolitician(entity models.Politician) Politician {
	return Politician{
		ID:       entity.ID,
		FullName: entity.FirstName + " " + entity.LastName,
	}
}
